components/finalizer/base/index.ts

```diff
@@ -0,0 +1,27 @@
+import conditions from '../conditions'
+import { ISpinParams } from '../../../types'
+
+/**
+ * @param {ISpinParams} params
+ */
+function check(params: ISpinParams): void {
+  const { settings, agentDI } = params
+  const { finalizerBase: config } = settings.finalizer
+  if (isEnable()) {
+    params.agentDI.injector.inject(params,{ contexts: ['1'] })
```

Comment:
Mixed paradigms: `inject` is a private member and should not be used directly.
This works because we assigned the object of Injector class.
This should be reworked.